model/proto/agent.proto

```diff
@@ -0,0 +1,13 @@
+syntax = "proto3";
+
+package elastic.apm.v1;
+
+option go_package = "/modelpb";
```

Comment:
> IIANM this should be the full import path, i.e. something like github.com/elastic/apm-data/model/modelpb

That would generate `github.com/elastic/apm-data/model/modelpb/*pb.go` in the destination directory as specified by https://protobuf.dev/getting-started/gotutorial/#compiling-protocol-buffers.

I'm currently (in a followup PR) generating the files in the `model/modelpb` folder so that would become `model/github.com/elastic/apm-data/model/modelpb/*pb.go`

I think google is storing them in a separate repo: https://github.com/googleapis/go-genproto

I would prefer not to create another deeply nested folder structure that would make it hard to import. 

WDYT ?

> Instead of repeating it in every file, could we specify it on the command line when invoking protoc, as mentioned at https://protobuf.dev/reference/go/go-generated/#package?

It's difficult to discuss so this is all code in a another PR but that would require to specify each file in the command line. From the page you linked: 
```
protoc --proto_path=src \
  --go_opt=Mprotos/buzz.proto=example.com/project/protos/fizz \
  --go_opt=Mprotos/bar.proto=example.com/project/protos/foo \
  protos/buzz.proto protos/bar.proto
```

I think adding and maintained the list of `.proto` file in the Makefile would be hard. I don't like repeating that line in each file but I couldn't find a better option. WDYT ?

Reply:
> That would generate github.com/elastic/apm-data/model/modelpb/*pb.go in the destination directory as specified by https://protobuf.dev/getting-started/gotutorial/#compiling-protocol-buffers.
>
> I'm currently (in a followup PR) generating the files in the model/modelpb folder so that would become model/github.com/elastic/apm-data/model/modelpb/*pb.go
>
>
> I would prefer not to create another deeply nested folder structure that would make it hard to import.

I think this works:

- in model/proto/*.proto, specify `option go_package = "github.com/elastic/apm-data/model/modelpb";`
- on the protoc command line, specify `protoc --go_out=. --go_opt=module=github.com/elastic/apm-data`

That says to write output files relative to the root of the repo, but trim the "github.com/elastic/apm-data" prefix off the generated file paths.

> It's difficult to discuss so this is all code in a another PR but that would require to specify each file in the command line. From the page you linked:

Ah sorry, I missed that you have to specify each file. That's annoying. Never mind then, just leave the `go_package` option in each file.

Reply:
Thanks for the guidance! :bow: 

It works! I've updated the code!